setuptools/tests/test_sdist.py

```diff
@@ -135,6 +134,47 @@ def test_package_data_in_sdist(self):
         assert os.path.join('sdist_test', 'c.rst') not in manifest
         assert os.path.join('d', 'e.dat') in manifest
 
+    def test_setup_py_exists(self):
+        dist = Distribution(SETUP_ATTRS)
+        dist.script_name = 'foo.py'
+        cmd = sdist(dist)
+        cmd.ensure_finalized()
+
+        with quiet():
+            cmd.run()
+
+        manifest = cmd.filelist.files
+        assert 'setup.py' in manifest
+
+    def test_setup_py_missing(self):
+        dist = Distribution(SETUP_ATTRS)
+        dist.script_name = 'foo.py'
+        cmd = sdist(dist)
+        cmd.ensure_finalized()
+
+        if os.path.exists("setup.py"):
+            os.remove("setup.py")
+        with quiet():
+            cmd.run()
+
+        manifest = cmd.filelist.files
+        assert 'setup.py' not in manifest
+
+    def test_setup_py_excluded(self):
+        with open("MANIFEST.in", "w") as manifest_file:
```

Comment:
@pganssle can it work without setup.py? shouldn't we break the build with an error if it's missing?

Reply:
From [PEP-517](https://www.python.org/dev/peps/pep-0517/)
> Where the build-backend key exists, it takes precedence over setup.py, and source trees need not include setup.py at all. Projects may still wish to include a setup.py for compatibility with tools that do not use this spec.

So there could be cases where someone has explicitly excluded from the Manifest and has a pyproject.toml

Reply:
We did talk about this, and I think it's fine. You have to affirmatively remove `setup.py`. At the moment I see no reason to do that, but as long as it's done explicitly I think we have to trust the user that they have a good reason.